Fix sudo password handling

CD3 opened the pull request:
First, thank you for pyinfra. I have been using pyinfra 2 to manage my personal dev machines after trying fabric and ansible, and really liked it, but with version 3's addition of `_if`, it is perfect for what I need.

The only issue I had when switching to 3.x was that the sudo password pompt no longer worked. Pyinfra would ask for the sudo password, but then immediately fail.

The problem was that the connector utility function that handled setting the sudo password was checking to see if `_sudo_password` was in the argument list, but not checking if it was set. Something like (i've shorted some var names)

```
a['_sudo_password'] = a.get( '_sudo_password', h.data.get('prompted_sudo_password')
```
but `a['_sudo_password']` is set to `None`, so the prompted password never get used.

The other issue was that the command string that was generated to pass the sudo password to an askpass executable was not quoting the password

```
PYINFRA_SUDO_PASSWORD=password
```

which does not work if the password contains special shell characters like `;' or spaces.

CD3 committed:
fix: quote password when setting PYINFRA_SUDO_PASSWORD

Commands that need to run with sudo are given the password by setting
the PYINFRA_SUDO_PASSWORD environment variable and using a script that
returns the value as the `askpass` executable.

example:

PYINFRA_SUDO_PASSWORD=p@ssworD sudo -H -A -k sh -c 'echo hi'

If the user's sudo password contains spaces or special characters that
are interpreted by the shell, then the password needs to be quoted

PYINFRA_SUDO_PASSWORD='p@ss worD' sudo -H -A -k sh -c 'echo hi'

CD3 committed:
fix: prompted sudo password is not ignored.

If the sudo password was needed, the user was prompted, but they
password they entered never ended up getting used because of a logic
error.

The prompted password would only be used if the _sudo_password argument
was missing. If no sudo password was given, it is set to None. So
_sudo_password is present in the arguents, but does not have a value.

This fix checks if the _sudo_password argument is None and sets it using
the prompted password if so.

pbrkr commented on pyinfra/connectors/util.py:
```diff
@@ -339,7 +338,7 @@ def make_unix_command(
                 "env",
                 "SUDO_ASKPASS={0}".format(_sudo_askpass_path),
-                MaskString("{0}={1}".format(SUDO_ASKPASS_ENV_VAR, _sudo_password)),
+                MaskString("{0}='{1}'".format(SUDO_ASKPASS_ENV_VAR, _sudo_password)),
```
I think `shlex.quote()` would be better here - that way it's safe even if the password includes single-quotes.

```
MaskString("{0}={1}".format(SUDO_ASKPASS_ENV_VAR, shlex.quote(_sudo_password))),
```

CD3 replied:
Oh yes, of course. I fixed it.

CD3 committed:
fix: use shelex.quote to quote sudo passwords

Need to use shelex.quote (rather than just single quoting) to handle
passwords that contain quotes.

Fizzadar reviewed (approved):
Fantastic stuff @CD3, thank you for fixing this AND adding a great test case 🎉
